ground_ir.c

```diff
@@ -28,6 +28,53 @@
 #define PULSE_L _LATD6
 #define PULSE_R _LATE4
 
+#define CALIB_HISTERES 20
+#define DEFAULT_CALIB 0x7FFF
+
+static unsigned char prox_calib_max_counter[2];
+static int prox_ground_max[2]; // the calibratoin is not store in settings
+
+static int _calib(int value, int i) {
+	int ret;
+	
+	if(value + CALIB_HISTERES > prox_ground_max[i]) {
+		if(++prox_calib_max_counter[i] > 3) {
+			if(value > prox_ground_max[i]) {
```

Comment:
If this is true, `prox_calib_max_counter[i]` is not set to 0, is it what you want?

Reply:
yes see lign 45 to 48. We want that 3 continuous value are over the max to update it.

Reply:
ok, good.